File: CP/CF/CF_EDU_99_div2_alt/4.py
import os, sys, bisect
from collections import defaultdict, Counter, deque;
from functools import lru_cache   #use @lru_cache(None)
if os.path.exists('in.txt'): sys.stdin=open('in.txt','r')
if os.path.exists('out.txt'): sys.stdout=open('out.txt', 'w')
#
input = lambda: sys.stdin.readline().strip()
imap = lambda: map(int,input().split()); ilist = lambda: list(imap())
#------------------------------------------------------------------
#sys.setrecursionlimit(10**6)
mod = int(1e9+7)

import sys
input = lambda: sys.stdin.readline().strip()
for _ in range(int(input())):
    n,x = map(int,input().split())
    a = list(map(int,input().split()))
    mx = max(x,max(a))
    dp = [[[-1]*(mx+1) for i in range(mx+1)] for j in range(n+1)]
    dp[0][0][x] = 0
    for i in range(n):
        for j in range(mx+1):
            for k in range(mx+1):
                if dp[i][j][k]==-1: continue
                if a[i]>k and k>=j:
                    if dp[i+1][k][a[i]]==-1:
                        dp[i+1][k][a[i]] = float("inf")
                    dp[i+1][k][a[i]] = min(dp[i+1][k][a[i]], dp[i][j][k]+1)
                if a[i]>=j:
                    if dp[i+1][a[i]][k]==-1:
                        dp[i+1][a[i]][k] = float("inf")
                    dp[i+1][a[i]][k] = min(dp[i+1][a[i]][k], dp[i][j][k])
    res = float("inf")
    for i in range(mx+1):
        for j in range(mx+1):
            if dp[n][i][j]==-1: continue
            res = min(dp[n][i][j], res)
    print(res if res!=float("inf") else -1)

File: CP/CF/CF_EDU_99_div2_alt/4.cpp
#include "bits/stdc++.h"
using namespace std;
#define ll long long

int dp[502][502][502];
int main(){
    // ios::sync_with_stdio(0),cin.tie(0);
    // freopen("in.txt", "r", stdin);
    int t = 1;
    cin >> t;
    while(t-->0){
                int n,x;
        cin >> n >> x;
        int a[n+1],mx=x;
        for(int i=1;i<=n;i++){
            cin >> a[i];
            mx=max(mx,a[i]);
        }
        // cout << n << '\n';
        int inf=2e9;
        // memset(dp,-1,sizeof dp);
        for(int i=0;i<=n;i++){
            for(int j=0;j<=mx;j++){
                for(int k=0;k<=mx;k++){
                    dp[i][j][k] = -1;
                }
            }
        }
         dp[0][0][x]=0;
         for(int i=0;i<n;i++){
            for(int j=0;j<=mx;j++){
                for(int k=0;k<=mx;k++){
                    if(dp[i][j][k]==-1)continue;
                    if(a[i+1]>k and k>=j){
                        if(dp[i+1][k][a[i+1]]==-1)dp[i+1][k][a[i+1]]=inf;
                        dp[i+1][k][a[i+1]]=min(dp[i+1][k][a[i+1]],dp[i][j][k]+1);
                    }
                    if(a[i+1]>=j){
                        if(dp[i+1][a[i+1]][k]==-1)dp[i+1][a[i+1]][k]=inf;
                        dp[i+1][a[i+1]][k]=min(dp[i+1][a[i+1]][k],dp[i][j][k]);
                    }
                }
            }
         }
        int ans=inf;
         for(int i=0;i<=mx;i++){
            for(int j=0;j<=mx;j++){
                if(dp[n][i][j]==-1)continue;
                ans=min(dp[n][i][j],ans);
        }
         }
         if(ans==inf)ans=-1;
         cout << ans << '\n';
    }
    return 0;
}
